packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -26,6 +26,10 @@ import '../vmservice.dart';
 
 const String protocolVersion = '0.2.0';
 
+/// Return whether the current zone is running in the context of daemon mode
+/// (i.e., being driven from an IDE vs the CLI).
+bool get isDaemonMode => logger is _AppRunLogger;
```

Comment:
Is there no better way to determine this? It feels a bit hacky that we determine what mode we're in from the logger.
Also, this (or rather, having a globally-public API here, and importing it from ../run_hot.dart) seems like a layering violation. Why would the run_hot.dart file know about commands?

Reply:
> Is there no better way to determine this? It feels a bit hacky that we determine what mode we're in from the logger.

It's indirect, but I'm ok with it as it's encapsulated in the library; people outside the lib don't need to know how it's implemented,

> Also, this (or rather, having a globally-public API here, and importing it from ../run_hot.dart) seems like a layering violation.

I agree - I didn't see a clear way to get the info to run_hot.dart - this file is the only one that knows that we're running in an IDE context vs a CLI one, and we want slightly different UI output for each.

I can think of ways to improve the abstraction here (by pushing data into the zone's context for instance), or land this as is; I would prefer to land but lmk if you have a preference.

Reply:
cc @tvolkert for ideas

Reply:
How about just making it an argument to the runner? Then the daemon command can set the argument appropriately when it creates it.

Reply:
> How about just making it an argument to the runner? Then the daemon command
> can set the argument appropriately when it creates it.

+1.  We definitely shouldn't introduce a dependency from the runner back to the command.  That also eliminates the need for the global public API.

Reply:
great, makes sense